# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/TrustedTypes.webidl
# Commit: 1ad85e47d90c
# Full Hash: 1ad85e47d90cd081bf4e8e4f5c674d8e8b22c501
# Author: Mirko Brodesser <mbrodesser@igalia.com>
# Date: 2024-06-20 16:10:57
# Regressor Bug: 1901491
# File Overlap Count: 1
# Description:
#   Bug 1901491: part 3) Implement `ShouldTrustedTypePolicyCreationBeBlockedByCSP` for Windows (not Workers) including reporting exceptions, excluding reporting violations. r=tschuster,webidl,saschanaz
#   
#   Parsing the trusted-types directive's values is not added to
#   `nsCSPParser::sourceList` because the latter supports keywords not
#   supported by the trusted-types directive. E.g. 'report-sample'.
# ==============================================================================

diff -r 8c535232b6c1 -r 1ad85e47d90c dom/webidl/TrustedTypes.webidl
--- a/dom/webidl/TrustedTypes.webidl	Thu Jun 20 07:57:31 2024 +0000
+++ b/dom/webidl/TrustedTypes.webidl	Thu Jun 20 07:57:31 2024 +0000
@@ -46,7 +46,7 @@
 
 [Exposed=(Window,Worker), Pref="dom.security.trusted_types.enabled"]
 interface TrustedTypePolicyFactory {
-    TrustedTypePolicy createPolicy(DOMString policyName , optional TrustedTypePolicyOptions policyOptions = {});
+    [Throws] TrustedTypePolicy createPolicy(DOMString policyName , optional TrustedTypePolicyOptions policyOptions = {});
     boolean isHTML(any value);
     boolean isScript(any value);
     boolean isScriptURL(any value);